### ISSUE TITLE
ERROR TypeError: Cannot read property 'skip' of undefined

### ISSUE THREAD

**Issue by nikksan** [valor-software/ng2-charts#992]
I am trying to dynamically add new data and rerender the chart accordingly, however  I get the following error:

> 72core.js:1598 ERROR TypeError: Cannot read property 'skip' of undefined
    at parseVisibleItems (core.interaction.js:39)
    at getIntersectItems (core.interaction.js:55)
    at indexMode (core.interaction.js:117)
    at Chart.getElementsAtEventForMode (core.controller.js:678)
    at Chart.handleEvent (core.controller.js:906)
    at Chart.eventHandler (core.controller.js:856)
    at listener (core.controller.js:793)
    at HTMLCanvasElement.proxies.(:4200/anonymous function) (http://localhost:4200/vendor.js:60521:4)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:4053)

Here is a the component
```
import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  // lineChart
  public lineChartData: Array<any> = [
    { data: [], label: 'Some random stuff..' }
  ];
  public lineChartLabels: Array<any> = [];
  public lineChartOptions: any = {
    responsive: true
  };
  public lineChartColors: Array<any> = [
    { // grey
      backgroundColor: 'rgba(148,159,177,0.2)',
      borderColor: 'rgba(148,159,177,1)',
      pointBackgroundColor: 'rgba(148,159,177,1)',
      pointBorderColor: '#fff',
      pointHoverBackgroundColor: '#fff',
      pointHoverBorderColor: 'rgba(148,159,177,0.8)'
    }
  ];
  public lineChartLegend: boolean = true;
  public lineChartType: string = 'line';

  ngOnInit() {
    setInterval(() => {
      this.lineChartData[0].data.push(Math.random())
      this.lineChartLabels.push((new Date).toUTCString())
    }, 1000)
  }
}
```
And the view:
```
<div class="row">
  <canvas baseChart width="400" height="400"
  [datasets]="lineChartData"
  [labels]="lineChartLabels"
  [options]="lineChartOptions"
  [colors]="lineChartColors"
  [legend]="lineChartLegend"
  [chartType]="lineChartType"
 ></canvas>
</div>  
```

**Comment by Joklogh** [comment]
Hi @nikksan
I copied your code and didn't get error. Try updating ng-charts as well as Angular, maybe it is problem in previous version.
Hope it will help, if not please let me know.

**Comment by jrahulroy** [comment]
+1

**Comment by paviad** [comment]
This issue stems from changing the underlying dataset without notifying the chart object by calling `update()` however previously, the `update()` method was hidden from clients, I have now exposed it through the `BaseChartDirective` and you may now call it manually.

Here's how your code should look with `ng2-charts@2.0.0-beta.9` :
```typescript
  @ViewChild(BaseChartDirective) chart: BaseChartDirective;
  ngOnInit() {
    setInterval(() => {
      this.lineChartData[0].data.push(Math.random());
      this.lineChartLabels.push((new Date).toUTCString());
      this.chart.update();
    }, 1000)
  }
```

Fixed by commit a77ec7c
